sources/editor/Stride.Editor/Engine/EntityExtensions.cs

```diff
@@ -20,12 +20,41 @@ public static Entity FindSubEntity(this Entity entity, Guid subEntityId)
             if (entity.Id == subEntityId)
                 return entity;
 
-            return entity.GetChildren().DepthFirst(x => x.GetChildren()).FirstOrDefault(x => x.Id == subEntityId);
+            foreach (var child in entity.Transform.Children)
+            {
+                if (child.Entity.FindSubEntity(subEntityId) is Entity e)
+                    return e;
+            }
+
+            return null;
         }
 
         public static Entity FindSubEntity(this Scene scene, Guid subEntityId)
         {
-            return scene.Entities.DepthFirst(x => x.GetChildren()).FirstOrDefault(x => x.Id == subEntityId);
+            if (scene.Entities.Count == 0)
+                return null;
+            
+            if (scene.Entities[0].EntityManager is EntityManager manager)
+            {
+                // The entity manager contains all entities of all the scenes under the root scene,
+                // it should be faster in most cases to iterate over that instead of going through the entity tree
```

Comment:
That path could be faster, but recursive calls should be avoided.

Reply:
There aren't any recursive calls here, just the one below which you commented on ? Tell me if I missed something here